<?xml version="1.0"?>
<launch>
    <arg name="range" default="5.0"/>
    <arg name="grid_num" default="50"/>
    <arg name="flow_image_size" default="50"/>
    <arg name="hz" default="100.0"/>
    <arg name="crop" default="5"/>
    <arg name="max_corners" default="20"/>
    <arg name="quality_level" default="0.05"/>
    <!-- <arg name="quality_level" default="0.0001"/> -->
    <arg name="min_distance" default="5.0"/>
    <arg name="win_size" default="10"/>
    <arg name="max_count" default="5"/>
    <arg name="is_save_image" default="false"/>
    <arg name="is_dense" default="false"/>
    <arg name="is_draw_flow_line" default="false"/>
    <arg name="save_data_num" default="1"/>
    <arg name="frame_id" default="odom"/>
    <arg name="child_frame_id" default="base_footprint"/>
    <arg name="step_border" default="2"/>
    <arg name="thickness" default="2"/>

	<param name="use_sim_time" value="false"/>

	<!-- <node pkg="bev_converter" type="ground_truth_odom_publisher" name="ground_truth_odom_publisher"> -->
	<!-- </node> -->

	<node pkg="bev_converter" type="bev_converter" name="bev_converter" output="screen">
		<param name="RANGE" value="$(arg range)" type="double"/>
		<param name="GRID_NUM" value="$(arg grid_num)" type="int"/>
		<param name="Hz" value="$(arg hz)" type="double"/>
		<param name="CHILD_FRAME_ID" value="$(arg child_frame_id)" type="str"/>
		<!-- <remap from="/odom" to="/ground_truth/odom"/> -->
	</node>

	<node pkg="bev_converter" type="bev_flow_estimator" name="bev_flow_estimator" output="screen">
		<param name="RANGE" value="$(arg range)" type="double"/>
		<param name="GRID_NUM" value="$(arg grid_num)" type="int"/>
		<param name="FLOW_IMAGE_SIZE" value="$(arg flow_image_size)" type="int"/>
		<param name="Hz" value="$(arg hz)" type="double"/>
		<param name="MANUAL_CROP_SIZE" value="$(arg crop)" type="int"/>
		<param name="MAX_CORNERS" value="$(arg max_corners)" type="int"/>
		<param name="QUALITY_LEVEL" value="$(arg quality_level)" type="double"/>
		<param name="MIN_DISTANCE" value="$(arg min_distance)" type="double"/>
		<param name="WIN_SIZE" value="$(arg win_size)" type="int"/>
		<param name="MAX_COUNT" value="$(arg max_count)" type="int"/>
		<param name="IS_SAVE_IMAGE" value="$(arg is_save_image)" type="bool"/>
		<param name="IS_DENSE" value="$(arg is_dense)" type="bool"/>
		<param name="IS_DRAW_FLOW_LINE" value="$(arg is_draw_flow_line)" type="bool"/>
		<param name="SAVE_NUMBER" value="$(arg save_data_num)" type="int"/>
		<param name="STEP_BORDER" value="$(arg step_border)" type="int"/>
		<param name="THICKNESS" value="$(arg thickness)" type="int"/>
		<param name="FRAME_ID" value="$(arg frame_id)" type="str"/>
		<param name="CHILD_FRAME_ID" value="$(arg child_frame_id)" type="str"/>
        <!-- <rosparam command="load" file="$(find cats)/config/cats_motion_param.yaml"/> -->
		<!-- <remap from="/odom" to="/ground_truth/odom"/> -->
	</node>

    <arg name="width" default="5.0"/>
    <arg name="resolution" default="0.10"/>
    <arg name="occupancy_threshold" default="0.5"/>
    <arg name="beam_num" default="720"/>
    <arg name="skip_scan_flag" default="false"/>

    <node pkg="dynamic_cloud_detector" type="dynamic_cloud_detector" name="dynamic_cloud_detector">
		<!-- <remap from="odom" to="/ground_truth/odom"/> -->
        <param name="WIDTH" value="$(arg width)"/>
        <param name="RESOLUTION" value="$(arg resolution)"/>
        <param name="OCCUPANCY_THRESHOLD" value="$(arg occupancy_threshold)"/>
        <param name="BEAM_NUM" value="$(arg beam_num)"/>
        <param name="SKIP_SCAN_FLAG" value="$(arg skip_scan_flag)"/>
        <remap from="/odom/complement" to="/odom"/>
    </node>

	<!-- <include file="$(find velodyne_height_map)/launch/amsl_velodyne_heightmap.launch"/> -->
	<!-- <include file="$(find velodyne_pointcloud)/launch/32e_points.launch"/> -->
</launch>
